kudos-services/src/main/java/org/exoplatform/kudos/listener/analytics/KudosSentListener.java

```diff
@@ -113,6 +113,15 @@ private void addEventStatistic(Kudos kudos) {
         }
       }
     }
+    // kudos sent for a user in a chosen audience
+    if (kudos.getSpacePrettyName() != null) {
+      Space space = getSpaceService().getSpaceByPrettyName(kudos.getSpacePrettyName());
+      Identity spaceIdentity = getIdentity(SpaceIdentityProvider.NAME, kudos.getSpacePrettyName());
+      if (spaceIdentity != null) {
+        streamIdentityId = Long.parseLong(spaceIdentity.getId());
+        addSpaceStatistics(statisticData, space);
+      }
```

Comment:
May be just adding space attributes when missing because `streamIdentityId` is already computed below :

```suggestion
      Space space = getSpaceService().getSpaceByPrettyName(kudos.getSpacePrettyName());
      if (space != null) {
        addSpaceStatistics(statisticData, space);
      }
```